src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -58,7 +58,11 @@ internal sealed class CommitManager : IAsyncCompletionCommitManager
                 return CommitResultUnhandled;
             }
 
-            var completionService = (CompletionServiceWithProviders)document.GetLanguageService<CompletionService>();
+            if (!(document.GetLanguageService<CompletionService>() is CompletionServiceWithProviders completionService))
+            {
+                return CommitResultUnhandled;
+            }
```

Comment:
i don't understand this... CompletionService is a public API that we've shipped.  What happens if someone has implemented and exported such a service?  This is basically bailing on commit for them.

Reply:
The comment was made when CommitManager casted CompletionService to CompletionServiceWithProviders and bailed out when it was not possible. Now CommitManager uses CompletionService only with no cast.

Reply:
The comment was made when CommitManager casted CompletionService to CompletionServiceWithProviders and bailed out when it was not possible. Now CommitManager uses CompletionService only with no cast.

---
In reply to: [234835414](https://github.com/dotnet/roslyn/pull/29016#discussion_r234835414) [](ancestors = 234835414)